### PR TITLE
WEB-568 TrackingConfig context provider

### PR DESCRIPTION
**Pull request by pladaria**

### CODE REVIEW

File: src/utils/analytics.tsx [hunk 1]
```diff
@@ -1,6 +1,31 @@
...
+};
+
+export const TrackingConfig = ({children, eventFormat}: TrackingProviderProps): JSX.Element => {
+    const value = React.useMemo(() => ({eventFormat}), [eventFormat]);
+
+    return <TrackingContext.Provider value={value}>{children}</TrackingContext.Provider>;
+};
+
+export const useTrackingConfig = (): {eventFormat?: EventFormat} => {
+    const {analytics} = useTheme();
```
Comment (atabel) [inline src/utils/analytics.tsx:27]: `ThemeContextProvider` could set a `TrackingConfig`

**Review by atabel — APPROVED** [review]

**Review by atabel — APPROVED** [review]

**Review by SergioMorchon — APPROVED** [review]

File: src/utils/analytics.tsx [hunk 2]
```diff
@@ -1,6 +1,31 @@
...
 export const eventActions = {linkTapped: 'link_tapped'} as const;
 
 // Google analytics 4
 export const eventNames = {userInteraction: 'user_interaction'} as const;
+
+const TrackingContext = React.createContext<{eventFormat?: EventFormat}>({eventFormat: undefined});
+
+type TrackingProviderProps = {
+    children: React.ReactNode;
+    eventFormat?: EventFormat;
```
[the remark this answers is not in the document]
Comment (pladaria) [reply src/utils/analytics.tsx]: changed

File: src/utils/analytics.tsx [hunk 3]
```diff
@@ -1,6 +1,31 @@
...
 // Universal Analytics (legacy)
 export const eventCategories = {userInteraction: 'user_interaction'} as const;
 export const eventActions = {linkTapped: 'link_tapped'} as const;
 
 // Google analytics 4
 export const eventNames = {userInteraction: 'user_interaction'} as const;
+
+const TrackingContext = React.createContext<{eventFormat?: EventFormat}>({eventFormat: undefined});
+
+type TrackingProviderProps = {
```
[the remark this answers is not in the document]
Comment (pladaria) [reply src/utils/analytics.tsx]: ok

File: src/utils/analytics.tsx (hunk 1 above)
Comment (pladaria) [reply src/utils/analytics.tsx]: done

**Review by RubenMorPov — APPROVED** [review]

**Pull request merged**